### DISCUSSION TITLE
How can I create MODData with my DataFrame?

### DISCUSSION THREAD

**Discussion by mhlee216** [ppdebreuck/modnet#63] [Q&A] [answered]
How can I create MODData with my DataFrame from my CSV file?
Or how can I append a DataFrame to featurized MODData (featurize()) ?

**Comment by ppdebreuck** [comment, marked as answer]
Hi @mhlee216,

You should first extract the structures (list of pymatgen structures or compositions) and targets (2D array) from the CSV file.
Example: you read the csv as a dataframe **df** with columns "structures" and "e_form".

```
data = MODData(materials = df["structures"],
               targets = df[["e_form"]].values,
               target_names = ["formation_energy"]
              )
```

If you already have features you can pass them to the `df_featurized` keyword. Otherwise use data.featurize().

Finally, you can append other features to the featurized MODData by using join or concat: `data.df_featurized = data.df_featurized.join(df_with_my_features)`. (Features are simply stored as a dataframe at _data.df_featurized_.)

Let me know if this works.

**Reply by mhlee216** [reply]
This works for me. Thank you so much for this solution!
